File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function ($scope, baseService) {


    /** 定义用户对象 */
    $scope.user = {};

    /** 定义保存用户方法 */
    $scope.save = function () {
        if ($scope.user.password != $scope.password) {
            alert("密码不一致，请重新输入！");
            return;
        }
        baseService.sendPost("/user/save?smsCode=" + $scope.smsCode,$scope.user)
            .then(function (response) {
                if(response.data) {
                    alert("注册成功！");
                    $scope.user = {};
                    $scope.password = "";
                    $scope.smsCode = "";
                } else {
                    alert("注册失败!")
                }
            })
    }
    
    
    /*定义发送短信验证码方法*/
    $scope.sendCode = function () {
        if($scope.user.phone) {
            baseService.sendGet("/user/sendCode?phone=" + $scope.user.phone)
                .then(function (response) {
                    alert(response.data?"发送成功" : "发送失败")
                })
        } else {

            alert("请输入手机号码")
        }
    }

});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.User;
import com.pinyougou.service.UserService;
import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 10000)
    private UserService userService;

    /*注册用户*/
    @PostMapping("/save")
    public boolean save(@RequestBody User user,String smsCode) {
        try {
            boolean ok = userService.checkSmsCode(user.getPhone(),smsCode);
            if(ok) {
            userService.save(user);
            }
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }


    /*发送短信验证码*/
    @GetMapping("/sendCode")
    public boolean sendCode(String phone) {
        try {
            if(StringUtils.isNoneBlank(phone)) {
                /*发送验证码*/
                userService.sendCode(phone);
                return true;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}
